addons/website/static/src/components/ace_warning/ace_warning.js

```diff
@@ -0,0 +1,41 @@
+/** @odoo-module **/
+
+import { EditHeadBodyDialog } from "../edit_head_body_dialog/edit_head_body_dialog";
+import { Component, useState } from "@odoo/owl";
+import { useService } from "@web/core/utils/hooks";
+
+export class AceEditorWarningOverlay extends Component {
+    static template = "website.AceEditorWarningOverlay";
+
+    setup() {
+        this.website = useService("website");
+        this.dialog = useService("dialog");
+
+        this.state = useState({
+            visible: (localStorage.getItem("website.ace.doNotShowWarning") || "false") === "false",
+        });
+
+        this.closeEditor = this.closeEditor.bind(this);
+        this.hideWarning = this.hideWarning.bind(this);
+        this.dontAskAgain = this.dontAskAgain.bind(this);
+        this.injectCode = this.injectCode.bind(this);
```

Comment:
You seem to be the only code in Odoo to do that to work with `t-on-click` so my guess is that this is not necessary. There also seem to be quite a followed convention to make handler function start with `on` (just like old Widget where the convention was `_on`).